src/configuration.ts

```diff
@@ -40,5 +43,13 @@ export default () => {
         any
     >;
     mergeEnvVariablesRecursive(config);
+
+    validate(plainToClass(Config, config)).then((errors) => {
+        for (const error of errors) {
+            console.error(error.toString());
+        }
+
+        if (errors.length > 0) throw new Error('Invalid config');
+    });
```

Comment:
I'm pretty sure that you should be able to just do something like
```ts
    const validatedConfig = plainToInstance(Config, config, {
        enableImplicitConversion: true,
    });
    const errors = validateSync(validatedConfig as object, {
        skipMissingProperties: false,
    });

    if (errors.length > 0) {
        throw new Error(errors.toString());
    }

    return validatedConfig;
```

I don't like chained callback (`.then()`)